osu.Game/Screens/Edit/Editor.cs

```diff
@@ -346,7 +348,7 @@ public override bool OnExiting(IScreen next)
 
         private void resetTrack(bool seekToStart = false)
         {
-            Beatmap.Value.Track?.Stop();
+            musicController.Stop();
```

Comment:
While I agree that stopping the track globally is probably not desired I don't think `IsUserPaused` should be exposed as it violates encapsulation. I'd rather have the editor resume the track if it stopped it on `OnExiting()` or otherwise.

Reply:
The problem here is that calling `MusicController.Play()` / `MusicController.Stop()` modifies `IsUserPaused` state, which is not an expected behaviour here and the name is misleading for what's gonna happen (the track is expected to be paused internally without the user being the reason of), I would say at the very least rename:
```csharp
MusicController.Play();
MusicController.Stop();
...
```
to
```csharp
MusicController.UserPlay();
MusicController.UserStop();
...
```
to avoid such an issue like the one above.

Reply:
While I agree this should be signalled better (and is perhaps maybe indicative of the fact that `MusicController` is mixing concerns too much) I still don't think exposing internal state is a proper solution to that. `User{Play,Stop}` would be ok with me I suppose.

Reply:
Just so we're all on the same page with the way I intended these changes to go down: the above change is a mishap from the way this PR took place - in two passes.

I intended for all non-test code of the form `beatmap.Track` to take on the form `music.CurrentTrack`, regardless of whether or not it is more correct than calling Play/Stop on `MusicController`. I'm trying to avoid logical changes as much as possible for this round of changes such that most such changes appear only in OsuGame/OsuGameBase/MusicController/TestScenes/BeatmapManager/WorkingBeatmap, which are either "overarching" components or ones that can't exist without change.